### PR TITLE
Queuing feedback

### PR DESCRIPTION
**Pull request by JustinPraas**
**Done:**
- Remove permission check on mount. Now displays a warning regarding permissions when appropriate with guidance
- Extract new History component
- Add history component to beamer mode and TA mode
- Add minute badges (with colouring) to queues
- Add colouring to number-of-entries-badge in queues
- Remove `newNotification` and `reminder` from application state
- Add `newNotification` and `reminder` to component states and sets them on server event
- Beamer mode reminders are 'queued' (pun inteded). Each reminder will display on the beamer for 3 seconds (**not thouroughly tested!**)

### CODE REVIEW

File: frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx [hunk 1]
```diff
@@ -253,17 +270,35 @@ class ProjectorQueuingPage extends Component<
...
                         mode={undefined}
                     />
                 ))}
+                <History />
             </Row>
         );
     }
+
+    private addReminderToQueue(reminder: RemindDto) {
+        const existingReminder = this.state.reminders.find(
```
Comment (harindup) [inline frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx]: Now, now, what do we know about race conditions in non-functional state updates that use the current state?

File: frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx [hunk 2]
```diff
@@ -253,17 +270,35 @@ class ProjectorQueuingPage extends Component<
...
+
+        if (existingReminder == null) {
+            const newReminders = this.state.reminders.slice();
+            newReminders.push(reminder);
+            this.setState(() => ({ reminders: newReminders }));
+        }
+    }
+
+    private removeReminderFromQueue() {
+        const newReminders = this.state.reminders.slice();
```
Comment (harindup) [inline frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx]: Ditto that state update comment

File: frontend/src/components/pages/queuing/Queue.tsx [hunk 3]
```diff
@@ -66,17 +74,20 @@ export default class Queue extends Component<QueueProps> {
...
-                                            outline
-                                            color="danger"
-                                            className="ml-2"
-                                            onClick={onDeleteQueue}
-                                        >
-                                            <FontAwesomeIcon icon={faTimes} />
-                                        </Button>
-                                    )}
+                                    {mode === QueuingMode.TA &&
+                                        onDeleteQueue != undefined && (
```
Comment (harindup) [inline frontend/src/components/pages/queuing/Queue.tsx]: No direct undefined checks, please

File: frontend/src/components/pages/queuing/Queue.tsx [hunk 4]
```diff
@@ -89,21 +100,22 @@ export default class Queue extends Component<QueueProps> {
...
-                                    >
-                                        <FontAwesomeIcon
-                                            className="mr-3"
-                                            icon={faCheck}
-                                        />
-                                        Accept next
-                                    </Button>
-                                )}
+                                {mode === QueuingMode.TA &&
+                                    onAcceptNext != undefined && (
```
Comment (harindup) [inline frontend/src/components/pages/queuing/Queue.tsx]: More undefined checks

File: frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx (hunk 1 above)
Comment (JustinPraas) [reply frontend/src/components/pages/queuing/ProjectorQueuingPage.tsx]: What would be your suggestion?

**Pull request merged**